### ISSUE TITLE
Not able to use valid FQDN as namespace in app mesh virtual nodes

### ISSUE THREAD

**Issue by gdowmont** [terraform-providers/terraform-provider-aws#9764]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.14
+ provider.aws v2.23.0

### Affected Resource(s)

* aws_appmesh_virtual_node

### Terraform Configuration Files

```hcl
resource "aws_appmesh_virtual_node" "virtual_node_blue" {
  name      = "${var.service_name}-blue"
  mesh_name = "${data.terraform_remote_state.service_mesh.mesh_id}"

  spec {
    listener {
      port_mapping {
        port     = "${var.service_port}"
        protocol = "http"
      }

      health_check {
        protocol            = "http"
        port                = "${var.service_port}"
        path                = "/api/home"
        unhealthy_threshold = 10
        healthy_threshold   = 2
        timeout_millis      = 2000
        interval_millis     = 5000
      }
    }

    service_discovery {
      aws_cloud_map {
        attributes = {
          ECS_TASK_DEFINITION_FAMILY = "${var.environment}-${var.service_name}-blue"
        }

        service_name   = "${var.service_name}"
        namespace_name = "example.domain.com"
      }
    }
  }
}
```

### Debug Output

```
Error: Error running plan: 1 error occurred:
        * module.nginx_two.aws_appmesh_virtual_node.virtual_node_green: only alphanumeric characters, underscores and hyphens allowed in "spec.0.service_discovery.0.aws_cloud_map.0.namespace_name"
```
### Expected Behavior

Setting this value through AWS Console is accepted so this also should be allowed through terraform.

### Actual Behavior

Validation does not allow dots (.) in the namespace name

### Steps to Reproduce

Setup an app mesh virtual node with cloud map discovery and FQDN as namespace name.

1. `terraform apply`

### Important Factoids
Validator `validateServiceDiscoveryHttpNamespaceName` does not allow dot (.) in regex. See https://github.com/terraform-providers/terraform-provider-aws/blob/c8049b7ba6b8a9a999966f76d6899ea0a6322a18/aws/validators.go#L2252-L2257

### References

* #0000

**Comment by ewbankkit** [comment]
The [API documentation](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_AwsCloudMapServiceDiscovery.html) says the pattern is:
```
((?=^.{1,127}$)^([a-zA-Z0-9_][a-zA-Z0-9-_]{0,61}[a-zA-Z0-9_]|[a-zA-Z0-9])(.([a-zA-Z0-9_][a-zA-Z0-9-_]{0,61}[a-zA-Z0-9_]|[a-zA-Z0-9]))*$)|(^.$)
```

Given that these regexes are in Java format and don't easily translate to Go format, I'll change this to just a 1-1024 length check (which is all the corresponding [Cloud Map API](https://docs.aws.amazon.com/cloud-map/latest/api/API_CreateHttpNamespace.html#API_CreateHttpNamespace_RequestParameters) specifies).

**Comment by gdowmont** [comment]
@ewbankkit Thanks for looking into it. That would definitely fix my issue.

**Comment by spaszek** [comment]
Is there any way to ignore this check for now?

**Comment by ewbankkit** [comment]
@spaszek No, we'll need https://github.com/terraform-providers/terraform-provider-aws/pull/9788 merged I think.

**Comment by bflad** [comment]
Regarding potential enhancements to ignoring validation, please see https://github.com/hashicorp/terraform-plugin-sdk/issues/105

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.39.0 of the Terraform AWS Provider, likely tomorrow. Thanks to @ewbankkit for the implementation. 👍
